app/controllers/taxa/names_controller.rb

```diff
@@ -53,14 +53,18 @@ def create
       current_user.add_agent unless current_user.agent
       agent = current_user.agent
       language = Language.find(params[:name][:synonym][:language_id])
-      synonym = @taxon_concept.add_common_name_synonym(params[:name][:string],
-                agent: agent, language: language, vetted: Vetted.trusted)
-      unless synonym.errors.blank?
-        flash[:error] = I18n.t(:common_name_exists, name_string: params[:name][:string])
+      if limit_names
+         flash[:error] = I18n.t(:error_user_limited)
       else
-        @taxon_concept.reindex_in_solr
-        log_action(@taxon_concept, synonym, :add_common_name)
-        expire_taxa([@taxon_concept.id])
+         synonym = @taxon_concept.add_common_name_synonym(params[:name][:string],
+                agent: agent, language: language, vetted: Vetted.trusted)
+         unless synonym.errors.blank?
```

Comment:
This won't work, as written. You probably meant

```
synonym = @taxon_concept.add_common_name_synonym(params[:name][:string],
                agent: agent, language: language, vetted: Vetted.trusted) unless
  synonym.errors.blank?
```

...The unless needed to be on the previous line.